ext/libdatadog_api/feature_flags.c

```diff
@@ -0,0 +1,366 @@
+#include <string.h>
+#include <stdlib.h>
+
+#include <ruby.h>
+
+#include <datadog/ffe.h>
+
+#include "datadog_ruby_common.h"
+
+// Forward declarations
+static VALUE configuration_new(VALUE self, VALUE json_str);
+static VALUE configuration_get_assignment(VALUE self, VALUE flag_key, VALUE context);
+static void configuration_free(void *ptr);
+
+static void evaluation_context_free(void *ptr);
+static VALUE evaluation_context_initialize_with_attributes(VALUE self, VALUE targeting_key, VALUE attributes_hash);
+
+static void resolution_details_free(void *ptr);
+
+// Resolution details accessor methods
+static VALUE resolution_details_get_value(VALUE self);
+static VALUE resolution_details_get_reason(VALUE self);
+static VALUE resolution_details_get_error_code(VALUE self);
+static VALUE resolution_details_get_error_message(VALUE self);
+static VALUE resolution_details_get_variant(VALUE self);
+static VALUE resolution_details_get_allocation_key(VALUE self);
+static VALUE resolution_details_get_do_log(VALUE self);
+
+
+
+void feature_flags_init(VALUE core_module) {
+  // Always define the Binding module - it will reuse existing if it exists
+  VALUE binding_module = rb_define_module_under(core_module, "FeatureFlags");
+
+  // Configuration class
+  VALUE configuration_class = rb_define_class_under(binding_module, "Configuration", rb_cObject);
+  rb_undef_alloc_func(configuration_class);
+  rb_define_singleton_method(configuration_class, "new", configuration_new, 1);
+  rb_define_method(configuration_class, "get_assignment", configuration_get_assignment, 2);
```

Comment:
I find it weird to have code like this

```ruby
def reconfigure!(configuration)
  @evaluator = Datadog::Core::FeatureFlags::Configuration.new(configuration)
end
```

because configuration implies a static thing unable to provide me anything except it's values